### PR TITLE
[GridDyna] Using expert filter for mapping

### PR DESCRIPTION
**Pull request by thangqp**
Depend on : https://github.com/gridsuite/commons-ui/pull/485

- Replace all UI components related to filter aspect by the expert filter component in commons-ui, i.e. CustomReactQueryBuilder
- Implement a mechanism in the FilterContainer to submit on-the-fly the react-hook-form  to redux, and validation is done in redux
- In addition,  rework on styles and re-organisation of UI components to make their size smaller

### CODE REVIEW

File: src/containers/RuleContainer.jsx [hunk 1]
```diff
@@ -140,94 +61,38 @@ const RuleContainer = ({ index, editParameters }) => {
...
+        (isAbsolute) => () =>
+            editParameters({
+                model: mappedModel,
+                setGroup,
+                groupType,
+                isAbsolute,
+                origin: GroupEditionOrigin.RULE,
+                originIndex: index,
+            }),
+        [editParameters, index]
```
Comment (Mathieu-Deharbe) [inline src/containers/RuleContainer.jsx:94]: [eslint]
 React Hook useCallback has missing dependencies: 'groupType', 'mappedModel', and 'setGroup'. Either include them or remove the dependency array

File: src/hooks/useFormOnUpdate.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import { useEffect } from 'react';
+import isDeepEqualReact from 'fast-deep-equal/react';
+import { UseFormReturn } from 'react-hook-form';
+
+const useFormOnUpdate = (formApi: UseFormReturn, data: any) => {
+    const prevData = usePrevious(data);
+
+    useEffect(() => {
+        if (!isDeepEqualReact(prevData, data)) {
+            const setValueToKey = ([key, value]: [string, any]) => {
```
Comment (Mathieu-Deharbe) [inline src/hooks/useFormOnUpdate.ts:18]: [eslint] 
 React Hook useEffect was passed a dependency list that is not an array literal. This means we can't statically verify whether you've passed the correct dependencies

 React Hook useEffect has missing dependencies: 'data' and 'formApi'. Either include them or remove the dependency array

**Review by Mathieu-Deharbe — CHANGES_REQUESTED** [review]
I didn't manage to test. I encountered various errors linked to the authentification (that I don't get when I test the main branch). Examples : 

- serializableStateInvariantMiddleware.ts:179 A non-serializable value was detected in an action, in the path: `payload.authenticationRouterError.userValidationError.error`. Value: TypeError: userManager.storeUser is not a function
- AuthenticationRouter.js:94 Uncaught TypeError: Cannot read properties of undefined (reading 'logoutError')
- The above error occurred in the <AuthenticationRouter> component:
    at AuthenticationRouter (http://localhost:3003/static/js/bundle.js:19288:3)
    
It may have something to do with the recent login updates in commons-ui, but i didn't investigate.

I also added the eslint warnings, but didn't review the whole code (and probably won't have the time to do it before my holidays)

File: src/containers/RuleContainer.jsx (hunk 1 above)
Comment (thangqp) [reply src/containers/RuleContainer.jsx]: DONE in https://github.com/gridsuite/griddyna-app/pull/95/commits/96f55374b90f2220f3662f8ecd54627790745aa2

File: src/hooks/useFormOnUpdate.ts (hunk 2 above)
Comment (thangqp) [reply src/hooks/useFormOnUpdate.ts]: I use ignore eslint : // eslint-disable-next-line react-hooks/exhaustive-deps, this is a particular case in which we force reset when mapping or equipment type switched

**Comment by thangqp** [comment]
> I didn't manage to test. I encountered various errors linked to the authentification (that I don't get when I test the main branch). Examples :
> 
>     * serializableStateInvariantMiddleware.ts:179 A non-serializable value was detected in an action, in the path: `payload.authenticationRouterError.userValidationError.error`. Value: TypeError: userManager.storeUser is not a function
> 
>     * AuthenticationRouter.js:94 Uncaught TypeError: Cannot read properties of undefined (reading 'logoutError')
> 
>     * The above error occurred in the  component:
>       at AuthenticationRouter (http://localhost:3003/static/js/bundle.js:19288:3)
> 
> 
> It may have something to do with the recent login updates in commons-ui, but i didn't investigate.
> 
> I also added the eslint warnings, but didn't review the whole code (and probably won't have the time to do it before my holidays)

Related to https://github.com/gridsuite/griddyna-app/pull/93

**Comment by Mathieu-Deharbe** [comment]
I know translation are not working in gridyna. Still that strange mix between english(should be mapped...), french(tension nominale poste), and what looks like missing translation keys (greaterOrEqual) is bad looking : 
image

This blue indicates a link : 
image
Using the same color/font that "Where" should be fine. (without upper case on the 'n')

**Review by Mathieu-Deharbe — DISMISSED** [review]
When I add a static var compensator "mapping" and click on the dropbox near "and use parameter group" I have the folowing error : 

console.js:288 MUI: You have provided an out-of-range value `[object Object]` for the select component.
Consider providing a value that matches one of the available options or ''.
The available values are `[object Object]`. 
    at SelectInput (http://localhost:3003/static/js/bundle.js:54479:27)

image

**Comment by thangqp** [comment]
> I know translation are not working in gridyna. Still that strange mix between english(should be mapped...), french(tension nominale poste), and what looks like missing translation keys (greaterOrEqual) is bad looking : image
> 
The translation is still in gridexplorer : https://github.com/gridsuite/gridexplore-app/blob/main/src/translations/en.json#L143-L148
They will be moved to commons-ui by a separated PR 


> This blue indicates a link : image Using the same color/font that "Where" should be fine. (without upper case on the 'n')
OK.. I will change font/color as the same of Where

**Review by etiennehomer — COMMENTED** [review]
Tests OK.
Improvement that can be done later: when mappings are updated, only the modified elements could be sent to the backend

**Comment by jonenst** [comment]
don't forget to revert package.json before merging

**Comment by thangqp** [comment]
> don't forget to revert package.json before merging

YES

**Comment by thangqp** [comment]
> Tests OK. Improvement that can be done later: when mappings are updated, only the modified elements could be sent to the backend

To discuss later about an appropriate solution

**Review by jonenst — APPROVED** [review]

**Comment by thangqp** [comment]
> When I add a static var compensator "mapping" and click on the dropbox near "and use parameter group" I have the folowing error :
> 
> console.js:288 MUI: You have provided an out-of-range value `[object Object]` for the select component. Consider providing a value that matches one of the available options or ''. The available values are `[object Object]`. at SelectInput (http://localhost:3003/static/js/bundle.js:54479:27)
> 
> image

This bug is out of scope this PR which is focus only on the change of filter system. Noted and will be correct in another PR.

**Pull request merged**